examples_tests/61.BlitFilterTest/main.cpp

```diff
@@ -0,0 +1,1561 @@
+		if (alphaSemantic == BlitFilter::CState::EAS_REFERENCE_OR_COVERAGE)
+		{
+			assert(alphaTestCounter);
+
+			// Memory dependency to ensure the alpha test pass has finished writing to alphaTestCounterBuffer
+			video::IGPUCommandBuffer::SBufferMemoryBarrier alphaTestBarrier = {};
+			alphaTestBarrier.barrier.srcAccessMask = asset::EAF_SHADER_WRITE_BIT;
+			alphaTestBarrier.barrier.dstAccessMask = asset::EAF_SHADER_READ_BIT;
+			alphaTestBarrier.srcQueueFamilyIndex = ~0u;
+			alphaTestBarrier.dstQueueFamilyIndex = ~0u;
+			alphaTestBarrier.buffer = alphaTestCounter->buffer;
+			alphaTestBarrier.size = alphaTestCounter->size;
+			alphaTestBarrier.offset = alphaTestCounter->offset;
+
+			// Memory dependency to ensure that the previous compute pass has finished writing to the output image
+			video::IGPUCommandBuffer::SImageMemoryBarrier readyForNorm = {};
+			readyForNorm.barrier.srcAccessMask = asset::EAF_SHADER_WRITE_BIT;
+			readyForNorm.barrier.dstAccessMask = static_cast<asset::E_ACCESS_FLAGS>(asset::EAF_SHADER_READ_BIT | asset::EAF_SHADER_WRITE_BIT);
+			readyForNorm.oldLayout = asset::EIL_GENERAL;
+			readyForNorm.newLayout = asset::EIL_GENERAL;
+			readyForNorm.srcQueueFamilyIndex = ~0u;
+			readyForNorm.dstQueueFamilyIndex = ~0u;
+			readyForNorm.image = outImage;
+			readyForNorm.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+			readyForNorm.subresourceRange.levelCount = 1u;
+			readyForNorm.subresourceRange.layerCount = 1u;
+			cmdbuf->pipelineBarrier(asset::EPSF_COMPUTE_SHADER_BIT, asset::EPSF_COMPUTE_SHADER_BIT, asset::EDF_NONE, 0u, nullptr, 1u, &alphaTestBarrier, 1u, &readyForNorm);
+
+			cmdbuf->bindDescriptorSets(asset::EPBP_COMPUTE, normalizationPipeline->getLayout(), 0u, 1u, &normalizationDS);
+			cmdbuf->bindComputePipeline(normalizationPipeline);
+			CBlitFilter::normalization_push_constants_t normPC = {};
+			CBlitFilter::dispatch_info_t normDispatchInfo = {};
+			buildNormalizationParameters(inImageExtent, outImageExtent, referenceAlpha, normPC, normDispatchInfo);
+			dispatchHelper<CBlitFilter::normalization_push_constants_t>(cmdbuf, normalizationPipeline->getLayout(), normPC, normDispatchInfo);
+		}
+	}
+
+	//! WARNING: This function blocks and stalls the GPU!
+	void blit(
+		video::IGPUQueue* computeQueue,
+		const BlitFilter::CState::E_ALPHA_SEMANTIC alphaSemantic,
+		video::IGPUDescriptorSet* alphaTestDS,
+		video::IGPUComputePipeline* alphaTestPipeline,
+		video::IGPUDescriptorSet* blitDS,
+		video::IGPUComputePipeline* blitPipeline,
+		video::IGPUDescriptorSet* normalizationDS,
+		video::IGPUComputePipeline* normalizationPipeline,
+		const core::vectorSIMDu32& inImageExtent,
+		const asset::IImage::E_TYPE inImageType,
+		const asset::E_FORMAT inImageFormat,
+		core::smart_refctd_ptr<video::IGPUImage> outImage,
+		const float referenceAlpha = 0.f,
+		const asset::SBufferRange<video::IGPUBuffer>* alphaTestCounter = nullptr)
```

Comment:
I'd use perfect forwarding for this (just declare the function inline as well)

Reply:
Resolved.